frontend/src/views/cluster/Overview/components/OverviewContent.vue

```diff
@@ -158,6 +160,27 @@ const toggleUseEmbeddedDiscoveryService = async (value: boolean) => {
   await setUseEmbeddedDiscoveryService(context.cluster ?? '', value)
 }
 
+const getComponentInRotation = computed(() => {
+  switch (secretRotationStatus.value?.spec.component) {
+    case 1:
+      return 'Talos CA'
+    default:
+      return ''
+  }
+})
+
+const secretRotationStatus: Ref<Resource<ClusterSecretsRotationStatusSpec> | undefined> = ref()
+const secretRotationStatusWatch = new Watch(secretRotationStatus)
+
+secretRotationStatusWatch.setup({
+  runtime: Runtime.Omni,
+  resource: {
+    namespace: DefaultNamespace,
+    type: ClusterSecretsRotationStatusType,
+    id: context.cluster,
+  },
+})
```

Comment:
lets stick to the simplified version of this I created:

```suggestion
const { data: secretRotationStatus } = useResourceWatch<ClusterSecretsRotationStatusSpec>(() => ({
  runtime: Runtime.Omni,
  resource: {
    namespace: DefaultNamespace,
    type: ClusterSecretsRotationStatusType,
    id: context.cluster,
  },
}))
```

> [!NOTE]
> IDK if `context.cluster` can change here but as a potentially reactive value i also put the watch params behind a getter function